CMakePresets.json

```diff
@@ -31,7 +31,10 @@
         "enable-sanitizers",
         "enable-targets-all",
         "enable-tests"
-      ]
+      ],
+      "cacheVariables": {
+        "OPENASSETIO_ENABLE_PYTHON_STUBGEN": "OFF"
```

Comment:
This is for CI.  It's annoying that there's no way to add code comments to JSON. From the commit message:

> Disable generation of stubs for the ASan CI job, since that would
require wrangling the `LD_PRELOAD` env var to allow stubgen to import
`_openassetio`, and going to that effort wouldn't give us much more
coverage, if any